src/spacel/provision/template/app_spot.py

```diff
@@ -21,13 +22,23 @@ def spotify(self, app, region, template):
             return
 
         resources = template['Resources']
-        self._add_spot_fleet(app, region, resources)
+        parameters = template['Parameters']
+        self._add_spot_fleet(app, region, resources, parameters)
         self._clean_up_asg(template)
 
     @staticmethod
-    def _add_spot_fleet(app, region, resources):
+    def _add_spot_fleet(app, region, resources, parameters):
         spot_price = app.spot.get('price', '1.00')
 
+        # Set Name tag
+        tags = {
+            'Name': {'Ref': 'AWS::StackName'}
+        }
+        user_data_param = parameters['UserData']['Default'] or ''
+        if user_data_param:
+            user_data_param += ','
+        user_data_param += '"tags":' + json.dumps(tags)
```

Comment:
Confused about behaviour here...
Does CloudFormation still resolve the `Ref` when it's inside a serialized string? My readthrough says instances are going to get `{'Ref': 'AWS::StackName'}` as their `Name` tag, not the actual stack name.


What I did in other places was leave a breadcrumb in the `.template` (i.e. https://github.com/pebble/spacel-provision/blob/master/src/spacel/cloudformation/elb-service.template#L507), then find and inject things (be they strings or Maps/Refs) after that breadcrumb: https://github.com/pebble/spacel-provision/blob/master/src/spacel/provision/db/cache.py#L83